app/util/analytics.py

```diff
@@ -19,10 +19,28 @@
 SERVICE_STARTED = 'SERVICE_STARTED'
 SUBJOB_EXECUTION_FINISH = 'SUBJOB_EXECUTION_FINISH'
 SUBJOB_EXECUTION_START = 'SUBJOB_EXECUTION_START'
+LOG_CACHE_MAX_SIZE = 10000
 
 _analytics_logger = None
 _eventlog_file = None
 _event_id_generator = Counter()
+_log_cache = collections.deque([], LOG_CACHE_MAX_SIZE)
```

Comment:
How about `collections.deque(maxlen=LOG_CACHE_MAX_SIZE)`?

Keyword args are one of my favorite things about Python. :neckbeard: